Generic script

hrvolapeter opened the pull request:
<!-- Please describe your changes on the following line: -->


---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `__` with appropriate data: -->
- [ ] `./mach build -d` does not report any errors
- [ ] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

<!-- Either: -->
- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

<!-- Also, please make sure that "Allow edits from maintainers" checkbox is checked, so that we can help you if you get stuck somewhere along the way.-->

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

hrvolapeter committed:
wip

paulrouget committed:
Refactor winit key handling

nox committed:
Remove outdated WebGL expectations

nox committed:
Fix program and shader lifetime cycle

felixrabe committed:
Fix copy paste typo

Commit:
Auto merge of #21281 - felixrabe:patch-1, r=paulrouget

Fix copy paste typo

<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/21281)
<!-- Reviewable:end -->

nox committed:
Remove an obsolete todo in gl.attachShader()

Attaching the same shader twice should indeed emit an InvalidOperation error.

nox committed:
Remove an obsolete todo in gl.compileShader()

We get the uniform data etc at link-time.

nox committed:
Simplify gl.getShaderParameter() (fixes #20562)

The expectation change is due to the shader now using its DOM-side compile status.
It is actually a bug for the shader to think it actually compiled successfully,
but at least it does so consistently now.

nox committed:
Properly check for context ownership of objects passed to gl.is*()

nox committed:
Fix some patched WebGL tests where the timeout was malformed

nox committed:
Add bug links to some WebGL expectations

nox committed:
Reenable glsl-long-variable-names.html

jdm committed:
Avoid frequent disk space errors on cross builders.

Commit:
Auto merge of #21250 - paulrouget:fixkeys, r=jdm

Refactor winit key handling

This should improve keys input on Linux and Windows.
Should fix #17146 and fix #21161

Tested Mac, Windows and Linux. Basic typing works, combo work, text navigation works. I hit some strange issues where sometimes the text would be displayed late, but I believe it is unrelated to this PR.

If we land that now, we will hit a regression on Mac, we need a winit update that includes https://github.com/tomaka/winit/pull/610.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/21250)
<!-- Reviewable:end -->

nox committed:
Mark many-draw-calls.html as long-running

nox committed:
Reenable origin-clean-conformance.html

jdm commented on components/script/dom/bindings/codegen/CodegenRust.py:
```diff
@@ -3182,6 +3182,21 @@ def definition_body(self):
+
+    def definition_body(self):
+        properties = PropertyArrays(self.descriptor)
+        define = ""
```
Since PropertyArrays contains a bunch of objects derived from PropertyDefiner, we can get the name of the relevant Rust array from the `variableNames` property. Then for each element in the array, we can replace the pointer member of the object with a name that is based on the corresponding entry in the `regular` list for that PropertyDefiner. Does that clear up the issue?

Commit:
Auto merge of #21286 - servo:jdm-patch-25, r=jdm

Avoid frequent disk space errors on cross builders.

This is killing our CI throughput. Fixes #21235.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/21286)
<!-- Reviewable:end -->

Commit:
Auto merge of #21284 - servo:webgl, r=emilio

Various fixes for the lifecycle of WebGL objects

<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/21284)
<!-- Reviewable:end -->
